### PR TITLE
App hints and reset button

### PR DESCRIPTION
**Pull request by ZayJackPrvy**
### Contributor checklist

-   [] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

I have created a new hint template that is displayed in `activity_main.xml`. Additionally, a reset functionality for the hints has been added, allowing the hints to be reset and displayed again as needed.

### Related issue

-   #166 #167

### CODE REVIEW

**Comment by andrewtavis** [comment]
Functionality wise all's looking good on my end :) Thanks for this, @ZayJackPrvy! I'll let @angrezichatterbox check the code a bit as well, and then we'll be good to merge 😊

File: app/src/main/java/be/scri/fragments/AboutFragment.kt [hunk 1]
```diff
@@ -168,6 +169,11 @@ class AboutFragment : Fragment() {
             ),
         )
 
+    private fun resetHints() {
+        HintUtils.resetHints(requireContext())
+        (activity as MainActivity).showHint("hint_shown_about", R.string.about_hint_message)
```
Comment (angrezichatterbox) [inline app/src/main/java/be/scri/fragments/AboutFragment.kt:174]: Could u replace this string with the localized strings available in string.xml and do the same for the other fragments' hint messages?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -10,6 +10,11 @@
     <string name="scribe_key">Scribe Key</string>
     <string name="command_bar">Command Bar</string>
 
+    <!-- Hints -->
```
Comment (angrezichatterbox) [inline app/src/main/res/values/strings.xml:13]: Once you replace the string you could remove these strings from the strings.xml

**Review by angrezichatterbox — COMMENTED** [review]
Thanks for the amazing PR @ZayJackPrvy. We are good to merge after these two changes.

File: app/src/main/java/be/scri/fragments/AboutFragment.kt (hunk 1 above)
Comment (ZayJackPrvy) [reply app/src/main/java/be/scri/fragments/AboutFragment.kt]: I've made the changes. Thanks for the review.

**Comment by andrewtavis** [comment]
Could you bring in the main branch and resolve the conflicts, @ZayJackPrvy, and then I'll bring this in next 😊

**Pull request closed without merging**